### PR TITLE
Add rule for Basic authentication #407

### PR DESCRIPTION
**Pull request by rolfedh**
Closes issue #407

### CODE REVIEW

File: .vale/styles/RedHat/CaseSensitiveTerms.yml [hunk 1]
```diff
@@ -12,6 +12,7 @@ swap:
   asciidoc|Asciidoc: AsciiDoc
   asciidoctor|AsciiDoctor: Asciidoctor
   '[nN]odejs|[nN]ode\.JS|node\.js': Node.js
+  basic authentication|Basic Authentication|basic auth|Basic auth|Basic Auth: Basic HTTP authentication (first instance) or Basic authentication
```
Comment (themr0c) [inline .vale/styles/RedHat/CaseSensitiveTerms.yml:15]: like `lower left|lower-left` in the TermsErrors rule:

```suggestion
  basic authentication|Basic Authentication|basic auth|Basic auth|Basic Auth: Basic HTTP authentication (first instance)|Basic authentication
```

File: .vale/styles/RedHat/TermsErrors.yml [hunk 2]
```diff
@@ -20,6 +20,7 @@ swap:
   "(?<!Mozilla )Thunderbird": Mozilla Thunderbird
   "a lot(?: of)?": many|much
   "backward(?:-)?compatible": compatible with earlier versions
+  "[bB]asic [aA]uth": Basic HTTP authentication (first instance) then Basic authentication
```
Comment (themr0c) [inline .vale/styles/RedHat/TermsErrors.yml:23]: like `lower left|lower-left` in the TermsErrors rule:

```suggestion
  "[bB]asic [aA]uth": Basic HTTP authentication (first instance)|Basic authentication
```

File: .vale/styles/RedHat/CaseSensitiveTerms.yml (hunk 1 above)
Comment (rolfedh) [reply .vale/styles/RedHat/CaseSensitiveTerms.yml:15]: Updated. Thanks.

File: .vale/styles/RedHat/TermsErrors.yml (hunk 2 above)
Comment (rolfedh) [reply .vale/styles/RedHat/TermsErrors.yml:23]: Thanks. Updated.

**Pull request merged**